src/couch_replicator_manager.erl

```diff
@@ -709,7 +708,27 @@ update_rep_doc(RepDbName, RepDocId, KVs, Wait) when is_binary(RepDocId) ->
             update_rep_doc(RepDbName, RepDocId, KVs, Wait * 2)
     end;
 update_rep_doc(RepDbName, #doc{body = {RepDocBody}} = RepDoc, KVs, _Try) ->
-    NewRepDocBody = lists:foldl(
+    PrevError = get_json_value(<<"_replication_state_reason">>, RepDocBody),
+    CurrentError = get_json_value(<<"_replication_state_reason">>, KVs),
+    case CurrentError =/= undefined andalso CurrentError =:= PrevError of
```

Comment:
ah then that ties in with the previous comment...user will still see "triggered" and undefined reason state at the same time